uefi-mkconfig

```diff
@@ -243,18 +194,90 @@ main () {
 		# Sort kernel images
 		kernel_images="$(sort -uV <<< "$kernel_images_old") $(sort -uV <<< "$kernel_images_new")"
 
-		# Add path back to the kernel image
-		for kernel_image in $kernel_images; do
-        		partition_efis_main+="$(find "$partition_mount" -name "$kernel_image") "
-		done
+	done 
+
+	# Add path back to the kernel image
+	for kernel_image in $kernel_images; do
+        	partition_efis+="$(find $partition_mounts -name "$kernel_image") "
+	done
+	
+	# Move backup entries to the beginning of the list
+	for efi_file in $partition_efis; do
+		if [[ -f "${efi_file}.uefibackup" ]]; then
+			partition_efis="${efi_file}.uefibackup ${partition_efis}"
+		fi
+	done
+	
+	for efi_file in $partition_efis; do
+		# Skip this kernel if ignore file is found
+		if [[ -f "$efi_file.ignore" ]]; then
+			ewarn "Ignoring \"$efi_file\" from partition \"$partition\"."
+			continue
+		fi
+
+		# Get partition mount
+		partition_mount="${efi_file/\/${efi_file#\/*\/}/}"
+
+		# Get partition
+		partition=$(mount | grep "$partition_mount ")
+		partition=${partition// */}
+		partition=${partition//*\//}
 		
-		# Create backup entry if it does not exist
-		partition_efis=$partition_efis_main
-		backup_entry
+		# Get partition label
+		partition_label="$(lsblk "/dev/$partition" -lno PARTLABEL)"
+
+		# Prepare path which will be inserted into efibootmgr
+		efi_file_path="${efi_file//$partition_mount}"
+
+		# Get kernel version
+		## Remove everything before /
+		kernel_version="${efi_file_path//*\//}"
+		## Remove everything before first - to remove any prefixes
+		kernel_version="${kernel_version/${kernel_version%%-*}-/}"
+		## Remove .efi suffix if it exists
+		if [[ "$kernel_version" == *".efi"* ]]; then
+			kernel_version="${kernel_version/.efi*/}"
+		fi
+			
+		# Check for existence of backup entry
+		if [[ "$efi_file" == *".uefibackup" ]]; then
+			if [[ "$initial_uefi_state" == *"UMCB ${kernel_version} on ${partition_label} "* ]]; then
+				einfo "Found already existing backup in ${partition}"
+				continue
+			elif [[ "$initial_uefi_state" == *"UMCB ${kernel_version} in ${partition_mount/\//} "* ]]; then
+				einfo "Found already existing backup in ${partition}"
+				continue
+			fi
+			backup_efi=0
+			efi_file="${efi_file/.uefibackup/}"
+		else
+			backup_efi=
```

Comment:
Instead of if/else, you can do here:
```
backup_efi=
if [[ .. ]]; then
    ....
    backup_efi=1
fi
```

Please also set it to 1 instead of 0, it doesn't make a functional difference since we check if it is empty or not, but in terms of readability "0" would usually translate to False, and "1" to True, so 1 would read more natural here.